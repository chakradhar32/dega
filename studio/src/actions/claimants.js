import axios from 'axios';
import {
  ADD_CLAIMANT,
  ADD_CLAIMANTS,
  ADD_CLAIMANTS_REQUEST,
  SET_CLAIMANTS_LOADING,
  RESET_CLAIMANTS,
  CLAIMANTS_API,
} from '../constants/claimants';
import { addErrorNotification, addSuccessNotification } from './notifications';
import { addMedia } from './media';
import getError from '../utils/getError';

export const getClaimants = (query) => {
  return (dispatch) => {
    dispatch(loadingClaimants());
    return axios
      .get(CLAIMANTS_API, {
        params: query,
      })
      .then((response) => {
        dispatch(
          addMedia(
            response.data.nodes
              .filter((claimant) => claimant.medium)
              .map((claimant) => claimant.medium),
          ),
        );
        dispatch(
          addClaimantsList(
            response.data.nodes.map((claimant) => {
              return { ...claimant, medium: claimant.medium?.id };
            }),
          ),
        );
        dispatch(
          addClaimantsRequest({
            data: response.data.nodes.map((item) => item.id),
            query: query,
            total: response.data.total,
          }),
        );
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopClaimantsLoading()));
  };
};

export const getClaimant = (id) => {
  return (dispatch) => {
    dispatch(loadingClaimants());
    return axios
      .get(CLAIMANTS_API + '/' + id)
      .then((response) => {
        if (response.data.medium) dispatch(addMedia([response.data.medium]));

        dispatch(getClaimantByID({ ...response.data, medium: response.data.medium?.id }));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopClaimantsLoading()));
  };
};

export const addClaimant = (data) => {
  return (dispatch) => {
    dispatch(loadingClaimants());
    return axios
      .post(CLAIMANTS_API, data)
      .then(() => {
        dispatch(resetClaimants());
        dispatch(addSuccessNotification('Claimant added'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};

export const updateClaimant = (data) => {
  return (dispatch) => {
    dispatch(loadingClaimants());
    return axios
      .put(CLAIMANTS_API + '/' + data.id, data)
      .then((response) => {
        if (response.data.medium) dispatch(addMedia([response.data.medium]));

        dispatch(getClaimantByID({ ...response.data, medium: response.data.medium?.id }));
        dispatch(addSuccessNotification('Claimant updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopClaimantsLoading()));
  };
};

export const deleteClaimant = (id) => {
  return (dispatch) => {
    dispatch(loadingClaimants());
    return axios
      .delete(CLAIMANTS_API + '/' + id)
      .then(() => {
        dispatch(resetClaimants());
        dispatch(addSuccessNotification('Claimant deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};

export const addClaimants = (claimants) => {
  return (dispatch) => {
    dispatch(
      addMedia(claimants.filter((claimant) => claimant.medium).map((claimant) => claimant.medium)),
    );
    dispatch(
      addClaimantsList(
        claimants.map((claimant) => {
          return { ...claimant, medium: claimant.medium?.id };
        }),
      ),
    );
  };
};

export const loadingClaimants = () => ({
  type: SET_CLAIMANTS_LOADING,
  payload: true,
});

export const stopClaimantsLoading = () => ({
  type: SET_CLAIMANTS_LOADING,
  payload: false,
});

export const getClaimantByID = (data) => ({
  type: ADD_CLAIMANT,
  payload: data,
});

export const addClaimantsList = (data) => ({
  type: ADD_CLAIMANTS,
  payload: data,
});

export const addClaimantsRequest = (data) => ({
  type: ADD_CLAIMANTS_REQUEST,
  payload: data,
});

export const resetClaimants = () => ({
  type: RESET_CLAIMANTS,
});
