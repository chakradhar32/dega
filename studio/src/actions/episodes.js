import axios from 'axios';
import {
  ADD_EPISODE,
  ADD_EPISODES,
  ADD_EPISODES_REQUEST,
  SET_EPISODES_LOADING,
  RESET_EPISODES,
  EPISODES_API,
} from '../constants/episodes';
import { addErrorNotification, addSuccessNotification } from './notifications';
import getError from '../utils/getError';
import { addPodcasts } from './podcasts';

export const getEpisodes = (query) => {
  return (dispatch) => {
    dispatch(loadingEpisodes());
    return axios
      .get(EPISODES_API, {
        params: query,
      })
      .then((response) => {
        dispatch(
          addPodcasts(
            response.data.nodes
              .filter((episode) => episode.podcast.id > 0)
              .map((episode) => {
                return episode.podcast;
              })
              .flat(1),
          ),
        );
        dispatch(
          addEpisodesList(
            response.data.nodes.map((episode) => {
              return { ...episode, podcast: episode.podcast.id };
            }),
          ),
        );
        dispatch(
          addEpisodesRequest({
            data: response.data.nodes.map((item) => item.id),
            query: query,
            total: response.data.total,
          }),
        );
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopEpisodesLoading()));
  };
};

export const getEpisode = (id) => {
  return (dispatch) => {
    dispatch(loadingEpisodes());
    return axios
      .get(EPISODES_API + '/' + id)
      .then((response) => {
        let episode = response.data;
        if (episode.podcast.id > 0) dispatch(addPodcasts([episode.podcast]));
        dispatch(getEpisodeByID({ ...episode, podcast: episode.podcast.id }));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopEpisodesLoading()));
  };
};

export const addEpisode = (data) => {
  return (dispatch) => {
    dispatch(loadingEpisodes());
    return axios
      .post(EPISODES_API, data)
      .then(() => {
        dispatch(resetEpisodes());
        dispatch(addSuccessNotification('Episode added'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};

export const updateEpisode = (data) => {
  return (dispatch) => {
    dispatch(loadingEpisodes());
    return axios
      .put(EPISODES_API + '/' + data.id, data)
      .then((response) => {
        let episode = response.data;
        if (episode.podcast.id > 0) dispatch(addPodcasts([episode.podcast]));
        dispatch(getEpisodeByID({ ...episode, podcast: episode.podcast.id }));
        dispatch(addSuccessNotification('Episode updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopEpisodesLoading()));
  };
};

export const deleteEpisode = (id) => {
  return (dispatch) => {
    dispatch(loadingEpisodes());
    return axios
      .delete(EPISODES_API + '/' + id)
      .then(() => {
        dispatch(resetEpisodes());
        dispatch(addSuccessNotification('Episode deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};

export const addEpisodes = (episodes) => {
  return (dispatch) => {
    dispatch(addEpisodesList(episodes));
  };
};

export const loadingEpisodes = () => ({
  type: SET_EPISODES_LOADING,
  payload: true,
});

export const stopEpisodesLoading = () => ({
  type: SET_EPISODES_LOADING,
  payload: false,
});

export const getEpisodeByID = (data) => ({
  type: ADD_EPISODE,
  payload: data,
});

export const addEpisodesList = (data) => ({
  type: ADD_EPISODES,
  payload: data,
});

export const addEpisodesRequest = (data) => ({
  type: ADD_EPISODES_REQUEST,
  payload: data,
});

export const resetEpisodes = () => ({
  type: RESET_EPISODES,
});
