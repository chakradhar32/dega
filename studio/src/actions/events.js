import axios from 'axios';
import {
  ADD_EVENTS,
  ADD_EVENTS_REQUEST,
  RESET_EVENTS,
  SET_EVENTS_LOADING,
  EVENTS_API,
  GET_EVENT,
  UPDATE_EVENT,
} from '../constants/events';
import { addErrorNotification, addSuccessNotification } from './notifications';
import getError from '../utils/getError';

// action to fetch default events
export const addDefaultEvents = (query) => {
  return (dispatch) => {
    dispatch(loadingEvents());
    return axios
      .post(EVENTS_API + '/default')
      .then((response) => {
        dispatch(addEvents(response.data.nodes));
        dispatch(
          addEventsRequest({
            data: response.data.nodes.map((item) => item.id),
            query: query,
            total: response.data.total,
          }),
        );
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopEventsLoading()));
  };
};

// action to fetch all events
export const getEvents = (query) => {
  return (dispatch) => {
    dispatch(loadingEvents());
    return axios
      .get(EVENTS_API, {
        params: query,
      })
      .then((response) => {
        dispatch(addEvents(response.data.nodes));
        dispatch(
          addEventsRequest({
            data: response.data.nodes.map((item) => item.id),
            query: query,
            total: response.data.total,
          }),
        );
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopEventsLoading()));
  };
};

// action to fetch event by id
export const getEvent = (id) => {
  return (dispatch) => {
    dispatch(loadingEvents());
    return axios
      .get(EVENTS_API + '/' + id)
      .then((response) => {
        dispatch(addEvent(GET_EVENT, response.data));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopEventsLoading()));
  };
};

// action to create event
export const createEvent = (data) => {
  return (dispatch) => {
    dispatch(loadingEvents());
    return axios
      .post(EVENTS_API, data)
      .then(() => {
        dispatch(resetEvents());
        dispatch(addSuccessNotification('Event created'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};

// action to update event by id
export const updateEvent = (data) => {
  return (dispatch) => {
    dispatch(loadingEvents());
    return axios
      .put(EVENTS_API + '/' + data.id, data)
      .then((response) => {
        dispatch(addEvent(UPDATE_EVENT, response.data));
        dispatch(addSuccessNotification('Event updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopEventsLoading()));
  };
};

// action to delete event by id
export const deleteEvent = (id) => {
  return (dispatch) => {
    dispatch(loadingEvents());
    return axios
      .delete(EVENTS_API + '/' + id)
      .then(() => {
        dispatch(resetEvents());
        dispatch(addSuccessNotification('Event deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};

export const loadingEvents = () => ({
  type: SET_EVENTS_LOADING,
  payload: true,
});

export const stopEventsLoading = () => ({
  type: SET_EVENTS_LOADING,
  payload: false,
});

export const addEvent = (type, payload) => ({
  type,
  payload,
});

export const addEvents = (payload) => ({
  type: ADD_EVENTS,
  payload,
});

export const addEventsRequest = (payload) => ({
  type: ADD_EVENTS_REQUEST,
  payload,
});

export const resetEvents = () => ({
  type: RESET_EVENTS,
});
