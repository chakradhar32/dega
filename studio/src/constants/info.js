export const ADD_INFO = 'ADD_INFO';
export const INFO_API = '/core/info';
