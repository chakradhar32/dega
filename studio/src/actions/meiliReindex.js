import axios from 'axios';
import { MEILI_REINDEX_API } from '../constants/meiliReindex';
import { addErrorNotification, addSuccessNotification } from './notifications';
import getError from '../utils/getError';

export const reindex = (id) => {
  return (dispatch) => {
    return axios
      .get(MEILI_REINDEX_API + '/' + id)
      .then((response) => {
        dispatch(addSuccessNotification('Successfully Reindexed'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};
export const reindexOrg = (orgId) => {
  return (dispatch) => {
    return axios
      .get(MEILI_REINDEX_API + '/' + orgId)
      .then((response) => {
        dispatch(addSuccessNotification('Successfully Reindexed'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};
