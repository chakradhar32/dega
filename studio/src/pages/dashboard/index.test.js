import React from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import { Provider, useDispatch } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import '../../matchMedia.mock';
import Dashboard from './index';
import { getInfo } from '../../actions/info';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));
jest.mock('../../actions/info', () => ({
  getInfo: jest.fn(),
}));
jest.mock('./components/Features', () => {
  const Features = () => <div />;
  return Features;
});

let store, mockedDispatch;
let state = {
  spaces: {
    orgs: [{ id: 1, organazation: 'Organization 1', spaces: [11] }],
    details: {
      11: { id: 11, name: 'Space 11', permissions: [{ resource: 'admin', actions: ['admin'] }] },
    },
    loading: false,
    selected: 11,
  },
  info: {
    categories: 5,
    episodes: 4,
    podcasts: 2,
    article: {
      draft: 38,
      template: 3,
      publish: 21,
      ready: 13,
    },
    factCheck: {
      draft: 31,
      template: 6,
      publish: 5,
      ready: 4,
    },
    posts: [],
    tags: 10,
  },
};

describe('Dashboard component', () => {
  describe('snapshot testing', () => {
    beforeEach(() => {
      store = mockStore({});
      store.dispatch = jest.fn();
      mockedDispatch = jest.fn();
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should render the component', () => {
      store = mockStore(state);
      const tree = mount(
        <Provider store={store}>
          <Router>
            <Dashboard />
          </Router>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
      expect(getInfo).toHaveBeenCalled();
    });
  });
  describe('component testing', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      mockedDispatch = jest.fn(() => new Promise((resolve) => resolve(true)));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should hide default feature option if not admin', () => {
      const state2 = { ...state };
      state2.spaces.details = {
        11: { id: 11, name: 'Space 11' },
      };
      const store2 = mockStore(state2);
      let wrapper;
      act(() => {
        wrapper = mount(
          <Provider store={store2}>
            <Router>
              <Dashboard />
            </Router>
          </Provider>,
        );
      });
    });
  });
});
