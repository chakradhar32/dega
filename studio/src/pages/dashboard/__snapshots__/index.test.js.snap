// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component snapshot testing should render the component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Dashboard>
        <Space
          direction="vertical"
        >
          <div
            className="ant-space ant-space-vertical"
            style={Object {}}
          >
            <Item
              className="ant-space-item"
              direction="vertical"
              index={0}
              key="ant-space-item-0"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={
                  Object {
                    "marginBottom": 8,
                  }
                }
              >
                <Title>
                  <Base
                    component="h1"
                  >
                    <LocaleReceiver
                      componentName="Text"
                    >
                      <ResizeObserver
                        disabled={true}
                        onResize={[Function]}
                      >
                        <Typography
                          className=""
                          component="h1"
                          style={
                            Object {
                              "WebkitLineClamp": undefined,
                            }
                          }
                        >
                          <h1
                            className="ant-typography"
                            style={
                              Object {
                                "WebkitLineClamp": undefined,
                              }
                            }
                          >
                            Dashboard
                          </h1>
                        </Typography>
                      </ResizeObserver>
                    </LocaleReceiver>
                  </Base>
                </Title>
              </div>
            </Item>
            <Item
              className="ant-space-item"
              direction="vertical"
              index={1}
              key="ant-space-item-1"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={
                  Object {
                    "marginBottom": 8,
                  }
                }
              >
                <Row
                  gutter={16}
                >
                  <div
                    className="ant-row"
                    style={
                      Object {
                        "marginLeft": -8,
                        "marginRight": -8,
                      }
                    }
                  >
                    <Col
                      span={6}
                    >
                      <div
                        className="ant-col ant-col-6"
                        style={
                          Object {
                            "paddingLeft": 8,
                            "paddingRight": 8,
                          }
                        }
                      >
                        <Link
                          to="/posts"
                        >
                          <LinkAnchor
                            href="/posts"
                            navigate={[Function]}
                          >
                            <a
                              href="/posts"
                              onClick={[Function]}
                            >
                              <withConfigConsumer(Statistic)
                                title="Total Posts"
                                value={72}
                              >
                                <Statistic
                                  decimalSeparator="."
                                  getPrefixCls={[Function]}
                                  groupSeparator=","
                                  loading={false}
                                  prefixCls="ant-statistic"
                                  renderEmpty={[Function]}
                                  title="Total Posts"
                                  value={72}
                                >
                                  <div
                                    className="ant-statistic"
                                  >
                                    <div
                                      className="ant-statistic-title"
                                    >
                                      Total Posts
                                    </div>
                                    <Skeleton
                                      avatar={false}
                                      loading={false}
                                      paragraph={false}
                                      title={true}
                                    >
                                      <div
                                        className="ant-statistic-content"
                                      >
                                        <StatisticNumber
                                          decimalSeparator="."
                                          getPrefixCls={[Function]}
                                          groupSeparator=","
                                          loading={false}
                                          prefixCls="ant-statistic"
                                          renderEmpty={[Function]}
                                          title="Total Posts"
                                          value={72}
                                        >
                                          <span
                                            className="ant-statistic-content-value"
                                          >
                                            <span
                                              className="ant-statistic-content-value-int"
                                              key="int"
                                            >
                                              72
                                            </span>
                                          </span>
                                        </StatisticNumber>
                                      </div>
                                    </Skeleton>
                                  </div>
                                </Statistic>
                              </withConfigConsumer(Statistic)>
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                    </Col>
                    <Col
                      span={6}
                    >
                      <div
                        className="ant-col ant-col-6"
                        style={
                          Object {
                            "paddingLeft": 8,
                            "paddingRight": 8,
                          }
                        }
                      >
                        <Link
                          to="/posts?status=publish"
                        >
                          <LinkAnchor
                            href="/posts?status=publish"
                            navigate={[Function]}
                          >
                            <a
                              href="/posts?status=publish"
                              onClick={[Function]}
                            >
                              <withConfigConsumer(Statistic)
                                title="Published Posts"
                                value={21}
                              >
                                <Statistic
                                  decimalSeparator="."
                                  getPrefixCls={[Function]}
                                  groupSeparator=","
                                  loading={false}
                                  prefixCls="ant-statistic"
                                  renderEmpty={[Function]}
                                  title="Published Posts"
                                  value={21}
                                >
                                  <div
                                    className="ant-statistic"
                                  >
                                    <div
                                      className="ant-statistic-title"
                                    >
                                      Published Posts
                                    </div>
                                    <Skeleton
                                      avatar={false}
                                      loading={false}
                                      paragraph={false}
                                      title={true}
                                    >
                                      <div
                                        className="ant-statistic-content"
                                      >
                                        <StatisticNumber
                                          decimalSeparator="."
                                          getPrefixCls={[Function]}
                                          groupSeparator=","
                                          loading={false}
                                          prefixCls="ant-statistic"
                                          renderEmpty={[Function]}
                                          title="Published Posts"
                                          value={21}
                                        >
                                          <span
                                            className="ant-statistic-content-value"
                                          >
                                            <span
                                              className="ant-statistic-content-value-int"
                                              key="int"
                                            >
                                              21
                                            </span>
                                          </span>
                                        </StatisticNumber>
                                      </div>
                                    </Skeleton>
                                  </div>
                                </Statistic>
                              </withConfigConsumer(Statistic)>
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                    </Col>
                    <Col
                      span={6}
                    >
                      <div
                        className="ant-col ant-col-6"
                        style={
                          Object {
                            "paddingLeft": 8,
                            "paddingRight": 8,
                          }
                        }
                      >
                        <Link
                          to="/posts?status=draft"
                        >
                          <LinkAnchor
                            href="/posts?status=draft"
                            navigate={[Function]}
                          >
                            <a
                              href="/posts?status=draft"
                              onClick={[Function]}
                            >
                              <withConfigConsumer(Statistic)
                                title="Draft Posts"
                                value={38}
                              >
                                <Statistic
                                  decimalSeparator="."
                                  getPrefixCls={[Function]}
                                  groupSeparator=","
                                  loading={false}
                                  prefixCls="ant-statistic"
                                  renderEmpty={[Function]}
                                  title="Draft Posts"
                                  value={38}
                                >
                                  <div
                                    className="ant-statistic"
                                  >
                                    <div
                                      className="ant-statistic-title"
                                    >
                                      Draft Posts
                                    </div>
                                    <Skeleton
                                      avatar={false}
                                      loading={false}
                                      paragraph={false}
                                      title={true}
                                    >
                                      <div
                                        className="ant-statistic-content"
                                      >
                                        <StatisticNumber
                                          decimalSeparator="."
                                          getPrefixCls={[Function]}
                                          groupSeparator=","
                                          loading={false}
                                          prefixCls="ant-statistic"
                                          renderEmpty={[Function]}
                                          title="Draft Posts"
                                          value={38}
                                        >
                                          <span
                                            className="ant-statistic-content-value"
                                          >
                                            <span
                                              className="ant-statistic-content-value-int"
                                              key="int"
                                            >
                                              38
                                            </span>
                                          </span>
                                        </StatisticNumber>
                                      </div>
                                    </Skeleton>
                                  </div>
                                </Statistic>
                              </withConfigConsumer(Statistic)>
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                    </Col>
                    <Col
                      span={6}
                    >
                      <div
                        className="ant-col ant-col-6"
                        style={
                          Object {
                            "paddingLeft": 8,
                            "paddingRight": 8,
                          }
                        }
                      >
                        <Link
                          to="/posts?status=ready"
                        >
                          <LinkAnchor
                            href="/posts?status=ready"
                            navigate={[Function]}
                          >
                            <a
                              href="/posts?status=ready"
                              onClick={[Function]}
                            >
                              <withConfigConsumer(Statistic)
                                title="Ready to publish Posts"
                                value={13}
                              >
                                <Statistic
                                  decimalSeparator="."
                                  getPrefixCls={[Function]}
                                  groupSeparator=","
                                  loading={false}
                                  prefixCls="ant-statistic"
                                  renderEmpty={[Function]}
                                  title="Ready to publish Posts"
                                  value={13}
                                >
                                  <div
                                    className="ant-statistic"
                                  >
                                    <div
                                      className="ant-statistic-title"
                                    >
                                      Ready to publish Posts
                                    </div>
                                    <Skeleton
                                      avatar={false}
                                      loading={false}
                                      paragraph={false}
                                      title={true}
                                    >
                                      <div
                                        className="ant-statistic-content"
                                      >
                                        <StatisticNumber
                                          decimalSeparator="."
                                          getPrefixCls={[Function]}
                                          groupSeparator=","
                                          loading={false}
                                          prefixCls="ant-statistic"
                                          renderEmpty={[Function]}
                                          title="Ready to publish Posts"
                                          value={13}
                                        >
                                          <span
                                            className="ant-statistic-content-value"
                                          >
                                            <span
                                              className="ant-statistic-content-value-int"
                                              key="int"
                                            >
                                              13
                                            </span>
                                          </span>
                                        </StatisticNumber>
                                      </div>
                                    </Skeleton>
                                  </div>
                                </Statistic>
                              </withConfigConsumer(Statistic)>
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                    </Col>
                  </div>
                </Row>
              </div>
            </Item>
            <Item
              className="ant-space-item"
              direction="vertical"
              index={2}
              key="ant-space-item-2"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={
                  Object {
                    "marginBottom": 8,
                  }
                }
              >
                <Row
                  gutter={16}
                >
                  <div
                    className="ant-row"
                    style={
                      Object {
                        "marginLeft": -8,
                        "marginRight": -8,
                      }
                    }
                  >
                    <Col
                      span={6}
                    >
                      <div
                        className="ant-col ant-col-6"
                        style={
                          Object {
                            "paddingLeft": 8,
                            "paddingRight": 8,
                          }
                        }
                      >
                        <Link
                          to="/fact-checks"
                        >
                          <LinkAnchor
                            href="/fact-checks"
                            navigate={[Function]}
                          >
                            <a
                              href="/fact-checks"
                              onClick={[Function]}
                            >
                              <withConfigConsumer(Statistic)
                                title="Total Fact Checks"
                                value={40}
                              >
                                <Statistic
                                  decimalSeparator="."
                                  getPrefixCls={[Function]}
                                  groupSeparator=","
                                  loading={false}
                                  prefixCls="ant-statistic"
                                  renderEmpty={[Function]}
                                  title="Total Fact Checks"
                                  value={40}
                                >
                                  <div
                                    className="ant-statistic"
                                  >
                                    <div
                                      className="ant-statistic-title"
                                    >
                                      Total Fact Checks
                                    </div>
                                    <Skeleton
                                      avatar={false}
                                      loading={false}
                                      paragraph={false}
                                      title={true}
                                    >
                                      <div
                                        className="ant-statistic-content"
                                      >
                                        <StatisticNumber
                                          decimalSeparator="."
                                          getPrefixCls={[Function]}
                                          groupSeparator=","
                                          loading={false}
                                          prefixCls="ant-statistic"
                                          renderEmpty={[Function]}
                                          title="Total Fact Checks"
                                          value={40}
                                        >
                                          <span
                                            className="ant-statistic-content-value"
                                          >
                                            <span
                                              className="ant-statistic-content-value-int"
                                              key="int"
                                            >
                                              40
                                            </span>
                                          </span>
                                        </StatisticNumber>
                                      </div>
                                    </Skeleton>
                                  </div>
                                </Statistic>
                              </withConfigConsumer(Statistic)>
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                    </Col>
                    <Col
                      span={6}
                    >
                      <div
                        className="ant-col ant-col-6"
                        style={
                          Object {
                            "paddingLeft": 8,
                            "paddingRight": 8,
                          }
                        }
                      >
                        <Link
                          to="/fact-checks?status=publish"
                        >
                          <LinkAnchor
                            href="/fact-checks?status=publish"
                            navigate={[Function]}
                          >
                            <a
                              href="/fact-checks?status=publish"
                              onClick={[Function]}
                            >
                              <withConfigConsumer(Statistic)
                                title="Published Fact Checks"
                                value={5}
                              >
                                <Statistic
                                  decimalSeparator="."
                                  getPrefixCls={[Function]}
                                  groupSeparator=","
                                  loading={false}
                                  prefixCls="ant-statistic"
                                  renderEmpty={[Function]}
                                  title="Published Fact Checks"
                                  value={5}
                                >
                                  <div
                                    className="ant-statistic"
                                  >
                                    <div
                                      className="ant-statistic-title"
                                    >
                                      Published Fact Checks
                                    </div>
                                    <Skeleton
                                      avatar={false}
                                      loading={false}
                                      paragraph={false}
                                      title={true}
                                    >
                                      <div
                                        className="ant-statistic-content"
                                      >
                                        <StatisticNumber
                                          decimalSeparator="."
                                          getPrefixCls={[Function]}
                                          groupSeparator=","
                                          loading={false}
                                          prefixCls="ant-statistic"
                                          renderEmpty={[Function]}
                                          title="Published Fact Checks"
                                          value={5}
                                        >
                                          <span
                                            className="ant-statistic-content-value"
                                          >
                                            <span
                                              className="ant-statistic-content-value-int"
                                              key="int"
                                            >
                                              5
                                            </span>
                                          </span>
                                        </StatisticNumber>
                                      </div>
                                    </Skeleton>
                                  </div>
                                </Statistic>
                              </withConfigConsumer(Statistic)>
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                    </Col>
                    <Col
                      span={6}
                    >
                      <div
                        className="ant-col ant-col-6"
                        style={
                          Object {
                            "paddingLeft": 8,
                            "paddingRight": 8,
                          }
                        }
                      >
                        <Link
                          to="/fact-checks?status=draft"
                        >
                          <LinkAnchor
                            href="/fact-checks?status=draft"
                            navigate={[Function]}
                          >
                            <a
                              href="/fact-checks?status=draft"
                              onClick={[Function]}
                            >
                              <withConfigConsumer(Statistic)
                                title="Draft Fact Checks"
                                value={31}
                              >
                                <Statistic
                                  decimalSeparator="."
                                  getPrefixCls={[Function]}
                                  groupSeparator=","
                                  loading={false}
                                  prefixCls="ant-statistic"
                                  renderEmpty={[Function]}
                                  title="Draft Fact Checks"
                                  value={31}
                                >
                                  <div
                                    className="ant-statistic"
                                  >
                                    <div
                                      className="ant-statistic-title"
                                    >
                                      Draft Fact Checks
                                    </div>
                                    <Skeleton
                                      avatar={false}
                                      loading={false}
                                      paragraph={false}
                                      title={true}
                                    >
                                      <div
                                        className="ant-statistic-content"
                                      >
                                        <StatisticNumber
                                          decimalSeparator="."
                                          getPrefixCls={[Function]}
                                          groupSeparator=","
                                          loading={false}
                                          prefixCls="ant-statistic"
                                          renderEmpty={[Function]}
                                          title="Draft Fact Checks"
                                          value={31}
                                        >
                                          <span
                                            className="ant-statistic-content-value"
                                          >
                                            <span
                                              className="ant-statistic-content-value-int"
                                              key="int"
                                            >
                                              31
                                            </span>
                                          </span>
                                        </StatisticNumber>
                                      </div>
                                    </Skeleton>
                                  </div>
                                </Statistic>
                              </withConfigConsumer(Statistic)>
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                    </Col>
                    <Col
                      span={6}
                    >
                      <div
                        className="ant-col ant-col-6"
                        style={
                          Object {
                            "paddingLeft": 8,
                            "paddingRight": 8,
                          }
                        }
                      >
                        <Link
                          to="/fact-checks?status=ready"
                        >
                          <LinkAnchor
                            href="/fact-checks?status=ready"
                            navigate={[Function]}
                          >
                            <a
                              href="/fact-checks?status=ready"
                              onClick={[Function]}
                            >
                              <withConfigConsumer(Statistic)
                                title="Ready to publish Fact Checks"
                                value={4}
                              >
                                <Statistic
                                  decimalSeparator="."
                                  getPrefixCls={[Function]}
                                  groupSeparator=","
                                  loading={false}
                                  prefixCls="ant-statistic"
                                  renderEmpty={[Function]}
                                  title="Ready to publish Fact Checks"
                                  value={4}
                                >
                                  <div
                                    className="ant-statistic"
                                  >
                                    <div
                                      className="ant-statistic-title"
                                    >
                                      Ready to publish Fact Checks
                                    </div>
                                    <Skeleton
                                      avatar={false}
                                      loading={false}
                                      paragraph={false}
                                      title={true}
                                    >
                                      <div
                                        className="ant-statistic-content"
                                      >
                                        <StatisticNumber
                                          decimalSeparator="."
                                          getPrefixCls={[Function]}
                                          groupSeparator=","
                                          loading={false}
                                          prefixCls="ant-statistic"
                                          renderEmpty={[Function]}
                                          title="Ready to publish Fact Checks"
                                          value={4}
                                        >
                                          <span
                                            className="ant-statistic-content-value"
                                          >
                                            <span
                                              className="ant-statistic-content-value-int"
                                              key="int"
                                            >
                                              4
                                            </span>
                                          </span>
                                        </StatisticNumber>
                                      </div>
                                    </Skeleton>
                                  </div>
                                </Statistic>
                              </withConfigConsumer(Statistic)>
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                    </Col>
                  </div>
                </Row>
              </div>
            </Item>
            <Item
              className="ant-space-item"
              direction="vertical"
              index={3}
              key="ant-space-item-3"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={Object {}}
              >
                <Features>
                  <div />
                </Features>
              </div>
            </Item>
          </div>
        </Space>
      </Dashboard>
    </Router>
  </BrowserRouter>
</Provider>
`;
